src/action/filesystems.rs

```diff
@@ -102,3 +165,92 @@ impl super::Response for Response {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+
+    use super::*;
+    use std::path::PathBuf;
+    use users::{get_current_uid, get_current_gid};
+
+    #[test]
+    fn test_is_any_filesystem_exist() {
+        let mut session = session::test::Fake::new();
+        assert!(handle(&mut session, ()).is_ok());
+        assert_ne!(session.reply_count(), 0);
+    }
+
+    /// Returns all responses whose `source` is equal to `fs_name`.
+    fn find_filesystems_by_name<'a>(
+        session: &'a session::test::Fake,
+        fs_name: &'a PathBuf,
+    ) -> Vec<&'a Response> {
+        let mut responses = Vec::new();
+
+        for i in 0..session.reply_count() {
+            let response: &Response = session.reply(i);
+            if response.mount_info.source == *fs_name {
+                responses.push(response);
+            }
+        }
+
+        responses
+    }
+
+    /// Unit-like struct, representing a filesystem for testing with `fuse`.
+    struct FuseFilesystem;
+
+    impl fuse::Filesystem for FuseFilesystem {}
+
+    #[test]
+    fn test_fuse_filesystem() {
+        let fs_name = PathBuf::from("fuse-test-fs");
+
+        let tmp_dir = tempfile::tempdir().unwrap();
+        let mountpoint = tmp_dir.path();
+
+        let fs_name_option = format!("fsname={}", fs_name.to_str().unwrap());
+        let options = [
+            "-o", "ro",
+            "-o", "nosuid",
+            "-o", "nodev",
+            "-o", "relatime",
+            "-o", &fs_name_option,
+        ];
+        let options = options.iter().map(|opt| opt.as_ref())
+            .collect::<Vec<&std::ffi::OsStr>>();
+
+        // Spawn a background thread to handle filesystem operations.
+        // When `_background_session` is dropped, filesystem will be unmounted.
+        let _background_session = unsafe {
```

Comment:
Instead of having an unused variable with `_` prefix to suppress warning about it, I prefer to add an explicit `drop(background_session)` at the end. This also ensures that you don't mistakenly drop it somewhere in between (because the compiler will not let you drop it twice).